### DISCUSSION TITLE
Can't Run Beacause No Spec file found Cypress

### DISCUSSION THREAD

**Discussion by priyadas2000** [cypress-io/cypress#26419] [Questions and Help]
I am using Cypress version 9.0.0 and I have created Product1 and Product2 in my integration folder.I am using headless mode to run testcases using command - 

`./node_modules/.bin/cypress run --spec cypress/integration/Product2/testcases/Login.spec.js`

my Directory structure is like:

    cypress/
       fixture/
       integration/
       Product1/
         testcases/
           Login.spec.js
           testlist   --->testlist contain fileName ex:Login.spec.js
         usecases/
         testlist     ---->testlist contain folderName ex:testcases usecases
         constant.js
       Product2/
         testcases/
           Login.spec.js
           testlist     ---->testlist contain fileName ex:Login.spec.js
         usecases/
         testlist       ---->testlist contain folderName ex:testcases usecases
         constant.js
      

getting error as-
Can't run because no spec files were found
We searched for specs matching this glob pattern:
 >/root/temp/abc/UI/cypress/integration/Product2/testcases/Login.spec.js

but if i used for Product1 ,
command
>/root/temp/abc/UI/cypress/integration/Product1/testcases/Login.spec.js

it Works fine and testcase run successfully.

I have Tried possibilities,
1. Changed in "specPattern":"cypress/integration/**/*.js" in cypress.json
2. Added testlist in integration which contain Product1 Product2 
3. Tried [github Discussion](https://github.com/cypress-io/cypress/issues/20734)
3. used 
`
npx cypress run --config integrationFolder=cypress/integation/Product2/testcases/Login.spec.js
`
after this,all testcases run for Product1

**Comment by portyu9** [comment]
Hi @priyadas2000 

Since you are using Cypress **9.0.0**, `specPattern` is not the correct configuration option. `specPattern` belongs to the Cypress 10+ configuration model.

Also, `integrationFolder` must point to a **directory**, not to an individual spec file.

The most useful clue is that the equivalent Product1 command works while Cypress reports that this exact Product2 path does not exist:

```text
/root/temp/abc/UI/cypress/integration/Product2/testcases/Login.spec.js
```

Check the filesystem from the same environment that runs Cypress:

```sh
pwd
ls -la cypress/integration/Product2/testcases/
test -f cypress/integration/Product2/testcases/Login.spec.js
find cypress/integration -iname 'login.spec.js' -print
```

Linux paths are case-sensitive, so differences such as `Product2` vs `product2`, `testcases` vs `testCases`, or `Login.spec.js` vs `login.spec.js` matter.

There is also a typo in the attempted configuration shown in the post:

```text
cypress/integation/...
```

instead of:

```text
cypress/integration/...
```

For Cypress 9, keep `integrationFolder` as the directory and use the old `testFiles` setting if you need to customize spec discovery.

If you upgrade to modern Cypress, the equivalent configuration is under `e2e.specPattern`.

Because an exact Product1 `--spec` path already works, I would resolve the Product2 filesystem/path mismatch before changing Cypress's spec-discovery configuration.

Please mark this answer as accepted if it helped, thank you.
